openapi.yml

```diff
@@ -107,5 +144,18 @@ components:
           $ref: "#/components/schemas/Address"
     OrderFull:
       allOf:
-        - $ref: "#/components/schemas/OrderNew"
+        - $ref: "#/components/schemas/OrderCreation"
         - $ref: "#/components/schemas/OrderExtra"
+    UUID:
+     description: |
+      Unique identifier for the order, calculated by keccak256(orderDigest, ownerAddress, validTo),
```

Comment:
In an ideal world, I think UUIDs should be opaque (as in it has no semantics to the user) and as such the clients wouldn't care how it gets calculated.

If we want to overload it with semantics, can we instead of hashing orderDigest and ownerAddress together, concatenate them with`_` or the like so that a person looking at the UUID can still recover the individual parts? Or is there value in making it 32 bytes exactly? 

Given that the orderDigest is needed for cancellation on the smart contract, it would be nice to have a way to request it without needing to do local js keccak magicery.

Reply:
Yeah, orderDigest and ownerAddress is also fine with me. I don't think there is a value in having it 32bytes exactly. If there is no resistance, I am happy to change it.

Reply:
Maybe it makes sense for it to be `orderDigest || ownerAddress || validTo` as 56-byte array as discussed https://github.com/gnosis/oba-contracts/pull/195#discussion_r533639378 so that It can be pasted directly into Etherscan?

Reply:
I would only go for `orderDigest || ownerAddress`, as this makes the UID unique already. But if you guys prefer nicks suggestion, please give him a thumbs up and I will change it.

Reply:
The idea behind using `orderDigest || ownerAddress || validTo` is that is allows this UID to be used to interact with the contract directly for reading fill amount and cancellation (on Etherscan for example). The hope was to reduce the amount of "magic numbers" used to identify orders to a single canonical form that can be used in multiple places.